### DISCUSSION TITLE
建议 eacpture 提供环境信息收集功能或者提供比较便利的收集环境信息的方式

### DISCUSSION THREAD

**Discussion by xxxxxliil** [gojue/ecapture#608] [Ideas]
当前，ecapture 提交错误报告时的 `Env: ` 一行需要提前 clone 下整个项目才能执行，不便于提交 bug 时附上环境信息

**Comment by cfc4n** [comment]
good idea.

**Comment by cfc4n** [comment]
我看了一下，在提交错误时，需要报告的env里，并不是`make env`命令输出的内容，这个是编译环境，并不是执行环境。

执行环境的话，在`ecapture tls`  等命令执行时，都已经在终端里打印出来了，只需要保存这些内容即可。

**Comment by xxxxxliil** [comment]
> 我看了一下，在提交错误时，需要报告的env里，并不是`make env`命令输出的内容，这个是编译环境，并不是执行环境。
> 
> 执行环境的话，在`ecapture tls` 等命令执行时，都已经在终端里打印出来了，只需要保存这些内容即可。

可以加一下 argv 输出和时间的毫秒吗，以及编译时把使用的 commit 次数和 commit 号也放进程序里。目前只有自行编译的 ecapture 有 commit 号记录，但是下载的并没有

**Comment by cfc4n** [comment]
运行时打印的版本信息中已经包括了。 比如`linux_arm64:0.7.7-20240303-bfb4a8c:5.15.0-113-generic`，含义为`Product OS`_`ARCH`:`Git Tags`-`Build Date`-`Commit ID`-`Builder Kernel Version`。

**Comment by xxxxxliil** [comment]
> 运行时打印的版本信息中已经包括了。 比如`linux_arm64:0.7.7-20240303-bfb4a8c:5.15.0-113-generic`，含义为`Product OS`_`ARCH`:`Git Tags`-`Build Date`-`Commit ID`-`Builder Kernel Version`。

原来如此，在 --version 中并未看到这部分就当作编译时没有加入，抱歉

**Comment by xxxxxliil** [comment]
@cfc4n 请问可以把这个 issue 转为 discussions 吗？因为有一部分需要讨论详细的方案（例如 debug 符号要不要分发，如何分发），同时当前 releases 页面的所有 ecapture 的版本信息都没有上面提到的 `Git Tags`-`Build Date`-`Commit ID`

**Reply by cfc4n** [reply]
在发行版里，保留的内容是`Product OS_ARCH`:`Git Tags`:`Builder Kernel Version`

`Git Tags`会对应唯一的commit ID，所以，我觉得不需要Commit ID了。

**Reply by xxxxxliil** [reply]
> 在发行版里，保留的内容是`Product OS_ARCH`:`Git Tags`:`Builder Kernel Version`
> 
> `Git Tags`会对应唯一的commit ID，所以，我觉得不需要Commit ID了。

了解

**Comment by xxxxxliil** [comment]
那目前没有的就是时间戳显示毫秒，和发行版的 debug 符号分发方式了。
个人希望加毫秒是为了便于区分日志间不足一秒但时间仍然较长的操作
目前因为没有 debug 符号不能确定 arch 系到底是什么东西导致的段错误

**Comment by cfc4n** [comment]
你是期望显示ecapture运行时，打印日志的毫秒级时间？

**Reply by xxxxxliil** [reply]
是的
